consensus/enclave/api/src/lib.rs

```diff
@@ -311,14 +328,41 @@ pub trait ConsensusEnclave: ReportableEnclave {
     /// Performs the first steps in accepting transactions from a remote client:
     /// 1) Re-encrypt all txs for the local enclave
     /// 2) Extract context data to be handed back to untrusted so that it could
-    /// collect the    information required by `tx_is_well_formed`.
+    /// collect the information required by `tx_is_well_formed`.
+    ///
+    /// Arguments:
+    /// * An EnclaveMessage which encrypts a `Tx` protobuf
+    ///
+    /// Returns:
+    /// * A Tx context including "locally encrypted" version of Tx and some extracted
+    ///   metadata or,
+    /// * A serialization / cryptography error.
     fn client_tx_propose(&self, msg: EnclaveMessage<ClientSession>) -> Result<TxContext>;
 
+    /// V2 API of client_tx_propose
+    ///
+    /// Arguments:
+    /// * An EnclaveMessage which encrypts a `ClientProposeTxRequest` protobuf
+    ///
+    /// Returns:
+    /// * A Tx context including "locally encrypted" version of Tx and some extracted
+    ///   metadata, or
+    /// * A serialization / cryptography error, or another mismatch error
+    ///   e.g. minimum_fee_map_hash or (some-day) chain-id error
+    fn client_tx_propose_v2(&self, msg: EnclaveMessage<ClientSession>) -> Result<TxContext>;
```

Comment:
we can't just fold these changes into `client_tx_propose` because we need to make a breaking change to the schema in order to insert another layer where we can add things. Making a new v2 version means that the old API continues to work and support clients when we drop in the new server.